### ISSUE TITLE
TF TransfoXL doesn't work with the `generate` method

### ISSUE THREAD

**Issue by LysandreJik** [huggingface/transformers#12651]
The TF TransfoXL model does not output `logits` but `prediction_scores` which are different due to the `AdaptiveEmbedding`.

The TF version of `generate` requires the `logits` to be output, therefore the model doesn't work with the `generate` method.
